### PR TITLE
Change run, execute command location

### PR DESCRIPTION
**Pull request by olivierodo**
## Change description

In order to get a better flexibility with the executor, let's move it at the hook level.
Since we are running the executor at the hook level now we can get access to the user config. 
From this user config we will be able to read the localhost port used by the user to run the server, then we also add a way to listen when the local server is ready to be tested by checking the port.

## Type of change
- [ ] Bug fix (fixes an issue)
- [x] New feature (adds functionality)

## Related issues

Indirected PR: https://github.com/restqa/http-mock-plugin/pull/3 

### Development

- [x] Lint rules pass locally
- [x] Application changes have been tested thoroughly
- [x] Automated tests covering modified code pass

### Security

- [x] Security impact of change has been considered
- [x] Code follows project security practices and guidelines

### Code review 

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [x] Changes have been reviewed by at least one other contributor
- [x] Pull request linked to issue tracker where applicable

### CODE REVIEW

File: src/core/plugin.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  if (!processor.BeforeAll || !processor.AfterAll) {
+    throw new Error(
+      "Please provide a processor containing the methods:  BeforeAll, AfterAll"
+    );
+  }
+
+  processor.Before(async function (scenario) {
+    // In order to fetch the dynamic data from extenral sources.
+    // Let's parse the scenario to get all the placeholder
+    await this.data.parse(scenario);
```
Comment (tony-go) [inline src/core/plugin.js:16]: Don't really understand this. Could you explain a bit more? 🙈

File: src/core/plugin.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  });
+
+  if (command) {
+    let server;
+    processor.BeforeAll(async function () {
+      if (typeof command === "string") {
+        const { mock } = this.restqa || {};
+        const { http } = mock || {}
+
+        server = await execute(command, http);
```
Comment (tony-go) [inline src/core/plugin.js:31]: The second argument should be `envs`, why we could have an `http` identifier here?

File: src/core/plugin.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        const restqapi = getPluginConfig("@restqa/restqapi", config);
+        if (!restqapi) {
+          throw new Error(Locale.get("error_missing_restqapi"));
+        }
+        const {url} = restqapi;
+        if (!url) {
+          throw new Error(Locale.get("error_missing_url"));
+        }
```
Comment (tony-go) [inline src/core/plugin.js:39]: Maybe should we make these check before executing: `server = await execute(command, http);`? WDYT?

File: src/core/plugin.test.js [hunk 4]
```diff
@@ -0,0 +1,532 @@
...
+      }).toThrow(
+        new Error(
+          "Please provide a processor containing the methods:  BeforeAll, AfterAll"
+        )
+      );
+    });
+
+    test("Throw error if the url is not found on the restqapi config", async () => {
+      const PORT = 5056;
+      const validRestQAConfigFile = `
```
Comment (tony-go) [inline src/core/plugin.test.js:56]:
```suggestion
      const configFileWithoutUrl = `
```

File: src/core/plugin.test.js [hunk 5]
```diff
@@ -0,0 +1,532 @@
...
+          port: 8888
+          envs:
+            GITHUB_API: github
+    outputs:
+      - type: file
+        enabled: true
+        config:
+          path: 'my-report.json'
+    `;
+      const configFile = jestqa.createTmpFile(
```
Comment (tony-go) [inline src/core/plugin.test.js:83]:
```suggestion
      const wrongConfigFile = jestqa.createTmpFile(
```

File: src/core/plugin.test.js [hunk 6]
```diff
@@ -0,0 +1,532 @@
...
+        new Error(
+          "Please share the target url of your service on @restqa/restqapi plugin section"
+        )
+      );
+      afterAll();
+    }, 10000);
+
+    test("Throw error if the restqapi plugin is not found", async () => {
+      const PORT = 5055;
+      const validRestQAConfigFile = `
```
Comment (tony-go) [inline src/core/plugin.test.js:141]:
```suggestion
      const configFileWithoutApiPlugin = `
```

File: src/core/plugin.test.js [hunk 7]
```diff
@@ -0,0 +1,532 @@
...
+          port: 8888
+          envs:
+            GITHUB_API: github
+    outputs:
+      - type: file
+        enabled: true
+        config:
+          path: 'my-report.json'
+    `;
+      const configFile = jestqa.createTmpFile(
```
Comment (tony-go) [inline src/core/plugin.test.js:165]:
```suggestion
      const wrongConfigFile = jestqa.createTmpFile(
```

**Review by tony-go — COMMENTED** [review]
Just wondering why didn't we use the plugin helper for this plugin? (probably an execution order issue)

File: src/core/plugin.js (hunk 1 above)
Comment (olivierodo) [reply src/core/plugin.js:16]: Will do offline

File: src/core/plugin.js (hunk 2 above)
Comment (olivierodo) [reply src/core/plugin.js:31]: This is to refer to the value from coming from the `http-mock-plugin`.

Comment (olivierodo) [reply src/core/plugin.js:31]: Updating it in order to increase the readability

**Review by tony-go — COMMENTED** [review]
Hello speed 😆 (jest options)

**Pull request merged**